app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt

```diff
@@ -82,28 +76,33 @@ class PlantDetailFragment : Fragment() {
         super.onCreateOptionsMenu(menu, inflater)
     }
 
-    @Suppress("DEPRECATION")
     override fun onOptionsItemSelected(item: MenuItem?): Boolean {
         return when (item?.itemId) {
             R.id.action_share -> {
-                val shareIntent = ShareCompat.IntentBuilder.from(activity)
-                    .setText(shareText)
-                    .setType("text/plain")
-                    .createChooserIntent()
-                    .apply {
-                        // https://android-developers.googleblog.com/2012/02/share-with-intents.html
-                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-                            // If we're on Lollipop, we can open the intent as a document
-                            addFlags(Intent.FLAG_ACTIVITY_NEW_DOCUMENT or Intent.FLAG_ACTIVITY_MULTIPLE_TASK)
-                        } else {
-                            // Else, we will use the old CLEAR_WHEN_TASK_RESET flag
-                            addFlags(Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET)
-                        }
-                    }
-                startActivity(shareIntent)
+                viewModel.share(requireContext())
                 return true
             }
             else -> super.onOptionsItemSelected(item)
         }
     }
+
+    @Suppress("DEPRECATION")
+    private fun onShare(shareText: String?) {
+        val text = shareText ?: ""
+        val shareIntent = ShareCompat.IntentBuilder.from(activity)
```

Comment:
I'm curious why this code couldn't mostly be in the ViewModel as well? That is, something like: `private fun onShare(shareText: String?) { startActivity(viewModel.buildShareIntent(shareText)) }`

Reply:
The `view-model` should from the aspect of architecture know nothing about the `activity`.